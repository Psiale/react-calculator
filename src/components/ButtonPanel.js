import React from 'react';
import Button from './Button';

const ButtonPanel = () => (
  <>
    <div>
      <Button name="AC" />
      <Button name="+/-" />
      <Button name="%" />
      <Button name="รท" />
    </div>

    <div>
      <Button name="7" />
      <Button name="8" />
      <Button name="9" />
      <Button name="X" />
    </div>
    <div>
      <Button name="4" />
      <Button name="5" />
      <Button name="6" />
      <Button name="-" />
    </div>
    <div>
      <Button name="1" />
      <Button name="2" />
      <Button name="3" />
      <Button name="-" />
    </div>
    <div>
      <Button name="0" />
      <Button name="." />
      <Button name="=" />
    </div>
  </>
);

export default ButtonPanel;

// Group 1: AC, +/-, %, รท
// Group 2: 7, 8, 9, X
// Group 3: 4, 5, 6, -
// Group 4: 1, 2, 3, +
// Group 5: 0, ., =
