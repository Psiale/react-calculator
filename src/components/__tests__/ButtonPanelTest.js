import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import ButtonPanel from '../ButtonPanel';

configure({ adapter: new Adapter() });
const mockfn = () => 'hello';

describe('ButtonPanel Component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<ButtonPanel clickHandler={mockfn} />).toJSON;
    expect(tree).toMatchSnapshot();
  });

  it('should have the name property displayed inner content ', () => {
    const wrapper = shallow(<ButtonPanel clickHandler={mockfn} />);
    const btn = wrapper.find('Button');
    expect(btn).toHaveLength(19);
  });
});
