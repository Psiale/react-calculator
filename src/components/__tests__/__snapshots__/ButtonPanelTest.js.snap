// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel Component matches the snapshot 1`] = `[Function]`;
